google/cloud/firestore_v1/stream_generator.py

```diff
@@ -50,10 +49,14 @@ def __iter__(self):
         return self
 
     def __next__(self):
-        next_value = self._generator.__next__()
-        if type(next_value) is query_profile_proto.ExplainMetrics:
-            self._explain_metrics = ExplainMetrics._from_pb(next_value)
-            return self._generator.__next__()
+        next_value, explain_metrics = self._generator.__next__()
+
+        if explain_metrics is not None:
+            self._explain_metrics = ExplainMetrics._from_pb(explain_metrics)
+
+            # Need to run the following iteration too, to ensure the length of
+            # the iteration isn't increased due to yielding explain_metrics.
+            return next(self)
```

Comment:
We can simplify this if we change the stream to yield both values together, right?